roles/customize_home/tasks/fix_env.yml

```
@@ -1,11 +1,11 @@
 - name: Check that the .env exists
   stat:
-    path: /home/vagrant/foreman/.env
+    path: "/home/{{ ansible_user }}/foreman/.env"
```

How about 
```suggestion
    path: "{{ ansible_user_dir }}/foreman/.env"
```
(from https://github.com/ansible/ansible/blob/stable-2.16/lib/ansible/module_utils/facts/system/user.py)

This should be present in any run that has facts enabled, but has the benefit of not guessing the home dir as `/home/$USERNAME` but actually looking it up from the user database.

---

Thanks, I'll give it a test

---

This only works if I manually add ansible_user_dir to the variables for the ansible provisioner in the box yaml config... otherwise (I guess because `become: true` in `playbooks/katello_devel.yml`) it still tries to use /root

I also realized that I was already overriding the value for ansible_user in my box, so that is likely the only reason my other approach was working (and I must have encountered something like this before already)

---

I suppose I could just add a role variable here with a default value that works for most setups to avoid this annoying behavior with `become`

---

ah, yeah… `become` certainly kills that.

role variable sounds good to me then!

---

Another idea, can we force `become: false` when running this role?

---

> Another idea, can we force `become: false` when running this role?

See the thread above this one that I marked as "resolved" :(

> but with `become: false` it fails due to permission denied.

---

> Another idea, can we force `become: false` when running this role?

although I was using `become: false` for just the one task and not the whole role, and the user should have permission to modify the contents of their own home directory....

---

OK, the issue appears to be that we have `become: true` in `playbooks/katello_devel.yml` and I wasn't able to override that

The good news is that it looks like `become` works for role dependencies in `meta/main.yml`, so I can remove it from `playbooks/katello_devel.yml` and add it back as needed per role dependency in `roles/katello_devel/meta/main.yml`